File: spring_java_orm_sample/src/main/java/com/example/demo/doma/dao/BookStockCustomDao.java
package com.example.demo.doma.dao;

import com.example.demo.doma.generator.entity.BookStock;
import org.seasar.doma.Dao;
import org.seasar.doma.Select;
import org.seasar.doma.boot.ConfigAutowireable;

@Dao
@ConfigAutowireable
public interface BookStockCustomDao {
    @Select
    BookStock selectByStoreIdAndBookIdWithWriteLock(Long bookStockStoreId, Long bookStockBookId) ;
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/doma/dao/BookStockCustomDao.kt
package com.example.demo.doma.dao

import com.example.demo.doma.generator.entity.BookStock
import org.seasar.doma.Dao
import org.seasar.doma.Select
import org.seasar.doma.boot.ConfigAutowireable

@Dao
@ConfigAutowireable
interface BookStockCustomDao {
    @Select
    fun selectByStoreIdAndBookIdWithWriteLock(bookStockStoreId: Long?, bookStockBookId: Long?): BookStock?
}
